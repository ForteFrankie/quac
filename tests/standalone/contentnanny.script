#!/bin/bash

# This script tests content issues. Currently, we ensure that all text-based
# files, except for certain excluded extensions, contain:
#
#   1. The copyright notice.
#   2. No tab characters.
#
# The output of this script is files which violate these guidelines; this is
# compared against the list of known exceptions in the .stdout file.

# Copyright (c) 2012-2013 Los Alamos National Security, LLC, and others

. ./environment.sh

cd ../..  # don't use $QUACBASE because it's an absolute path

C='Copyright (\(c\)|Â©) 2012-2013 Los Alamos National Security, LLC, and others'

# We use find rather than git ls-files so that we catch problems even in files
# not yet added to git. We skip some directories:
#
#   .git                  Git metadata
#   deps                  dependencies that are not part of QUAC
#   doc                   compiled documentation
#   sphinx/_build         compiled documentation
#   tests/twitter/tweets  mostly Twitter data that are not always present
for filename in $(find . \( -path ./.git -o -path ./deps -o -path ./doc -o -path ./sphinx/_build -o -path ./tests/twitter/tweets \) -prune -o -type f | LC_ALL=C sort); do
    if (file $filename | fgrep -q 'text' \
        && echo $filename | egrep -qv '\.(gz|stderr|stdout|tmp|tsv)'); then

        # Missing copyright notices.
        egrep -iL "$C" $filename | sed 's/^/copyright missing: /'

        # Tabs. Specifically, for each file containing at least one tab,
        # prints the number of lines containing a tab. Finding these lines is
        # up to you. :) (We don't print line numbers because that will change
        # based on non-tab revisions.)
        if (echo $filename | egrep -qv '(\.gz$|/Makefile$|\.mk$)'); then

            fgrep -Hc '	' $filename | egrep -v ':0$' | sed 's/^/tab(s) present:    /'
        fi

    fi
done
