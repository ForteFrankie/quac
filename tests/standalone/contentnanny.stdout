copyright missing: ./.gitignore
copyright missing: ./COPYRIGHT.rst
copyright missing: ./LICENSE
copyright missing: ./README.rst
tab(s) present:    ./bin/tsvgrep:1
copyright missing: ./lib/disco/README
tab(s) present:    ./lib/qr/scripting.py:6
tab(s) present:    ./lib/u.py:1
copyright missing: ./lib/unicodedata2.py
copyright missing: ./requirements.txt
copyright missing: ./sphinx/Makefile
copyright missing: ./sphinx/conf.py
tab(s) present:    ./sphinx/map_reduce.rst:1
copyright missing: ./tests/README
tab(s) present:    ./tests/standalone/contentnanny.script:1
tab(s) present:    ./tests/standalone/hashsplit.script:22
tab(s) present:    ./tests/standalone/quacreduce.script:1
copyright missing: ./tests/standalone/wp-access/README
copyright missing: ./tests/standalone/wp-access/ls-R.fulldata
copyright missing: ./tests/tmp/README
copyright missing: ./tests/twitter/README
